src/app/upload/shared/audio-cancel.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive, Input, HostListener } from '@angular/core';
+import { AudioFileModel, AudioVersionModel } from '../../shared';
+
+@Directive({
+  selector: '[publishAudioCancel]'
+})
+export class AudioCancelDirective {
+
+  @Input() delay = 1000;
+
+  @Input() publishAudioCancel: AudioFileModel;
+
+  @Input() version: AudioVersionModel;
+
+  @HostListener('click') onClick() {
+    this.publishAudioCancel.canceled = true;
+    setTimeout(() => {
+      this.publishAudioCancel.destroy();
+      this.version.removeUpload(this.publishAudioCancel);
+      this.publishAudioCancel.canceled = false;
+    }, this.delay);
```

Comment:
This transition is a very nice touch. :) Also, hmm so that's how defaults are done, makes so much sense (still so much to learn, big and small.)